### PR TITLE
Use `globalThis` for `global`

### PR DESCRIPTION
**Pull request by JeanMeche**
use `globalThis` for `global`

`globalThis` is now available on every runtime supported by Angular

See individual commits for the 3 impacted packages. 

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [x] No

### CODE REVIEW

**Comment by dylhunn** [comment]
I have no idea whether this is correct, and I'm honest not sure who to ask. Maybe @devversion?

**Comment by alfaproject** [comment]
Isn't globalThis supported by all the browsers that Angular supports now?

**Comment by JeanMeche** [comment]
@alfaproject You're making a goodpoint, the global definitions could be simplified now that IE has been dropped.

**Comment by JeanMeche** [comment]
@dylhunn Following @alfaproject's comment, I change the scope of the PR to actually generalize the behaviour !

File: packages/localize/src/localize/src/global.ts [hunk 1]
```diff
@@ -9,28 +9,4 @@
...
-// environments, then fallback to __global first, because in Node tests both __global and
-// __window may be defined and _global should be __global in that case. Note: Typeof/Instanceof
-// checks are considered side-effects in Terser. We explicitly mark this as side-effect free:
-// https://github.com/terser/terser/issues/250.
-export const _global: any = (/* @__PURE__ */ (
-    () => (typeof globalThis !== 'undefined' && globalThis) ||
-        (typeof global !== 'undefined' && global) || (typeof window !== 'undefined' && window) ||
-        (typeof self !== 'undefined' && typeof WorkerGlobalScope !== 'undefined' &&
-         self instanceof WorkerGlobalScope && self))());
+export const _global: any = globalThis;
```
Comment (JeanMeche) [inline packages/localize/src/localize/src/global.ts:12]: I was thinking about deleting the file. 
It only imported by `localize/init` to affect `$localize` to the global scope. 

wdyt ?

File: packages/core/testing/src/test_hooks.ts [hunk 2]
```diff
@@ -15,21 +15,13 @@
...
 
-declare var global: any;
-
-const _global = <any>(typeof window === 'undefined' ? global : window);
-
 // Reset the test providers and the fake async zone before each test.
-if (_global.beforeEach) {
-  _global.beforeEach(getCleanupHook(false));
-}
+globalThis.beforeEach(getCleanupHook(false));
```
Comment (alfaproject) [inline packages/core/testing/src/test_hooks.ts:19]: I'm guessing that the beforeEach/forEach don't need to `globalThis.` anymore, no?

(same in other places)

**Review by devversion — APPROVED** [review]
Seems reasonable to me. I think the current logic is mostly a historic artifact

**Comment by AndrewKushnir** [comment]
@alan-agius4 do you see any risks from SSR standpoint?

**Review by alan-agius4 — APPROVED** [review]
LGTM

**Comment by AndrewKushnir** [comment]
[TGP](https://test.corp.google.com/OCL:545581332:BASE:545580575:1688540989208:c5327fff).

**Review by AndrewKushnir — APPROVED** [review]

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 67df5a977170575e4a59dd477dbf3051cd92e1ba.

**Pull request closed without merging**